src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -87,20 +108,38 @@ private void Commit(Completion completion)
 
             Text = completion.CompletionItem.InsertionText;
             LibrarySearchBox.CaretIndex = Text.IndexOf(completion.CompletionItem.DisplayText, StringComparison.OrdinalIgnoreCase) + completion.CompletionItem.DisplayText.Length;
+            Flyout.IsOpen = false;
+            SelectedItem = null;
         }
 
         private void HandleKeyPress(object sender, KeyEventArgs e)
         {
             switch (e.Key)
             {
                 case Key.Tab:
-                case Key.Enter:
+                    // SelectedItem could be null if the key press came from keyboard navigation and not commit operation.
+                    // In this case we will just move the focus to next control.
+                    if (SelectedItem == null)
+                    {
+                        LibrarySearchBox.MoveFocus(new TraversalRequest(FocusNavigationDirection.Next));
+                        e.Handled = true;
```

Comment:
>e.Handled = true; [](start = 24, length = 17)

Done in both cases

Reply:
Cleaned this up in iteration 8.

---
In reply to: [200471970](https://github.com/aspnet/LibraryManager/pull/278#discussion_r200471970) [](ancestors = 200471970)